iavl/iavl_proof2.go

```diff
@@ -133,15 +54,15 @@ func (proof *KeyAbsentProof) Verify(key []byte, root []byte) error {
 		if err := proof.LeftPath.verify(proof.LeftNode, root); err != nil {
 			return errors.New("failed to verify left path")
 		}
-		if bytes.Compare(proof.LeftNode.KeyBytes, key) != -1 {
+		if !proof.LeftNode.isLesserThan(key) {
```

Comment:
Hmm I think "lesser" is more correct. Greater and lesser are both adjectives and the other function is `isGreaterThan`.